components/cache/cache_pools.rst

```diff
@@ -0,0 +1,243 @@
+can keep using the same cache pool even if the underlying cache mechanism
+changes from a file system based cache to a Redis or database based cache.
+
+Creating Cache Pools
+--------------------
+
+Cache Pools are created through the **cache adapters**, which are classes that
+implement the :class:`Psr\\Cache\\CacheItemPoolInterface` interface. This
+component provides several adapters ready to use in your applications.
+
+Array Cache Adapter
+~~~~~~~~~~~~~~~~~~~
+
+This adapter is only useful for testing purposes because contents are stored in
+memory and not persisted in any way. Besides, some features explained later are
+not available, such as the deferred saves::
+
+    use Symfony\Component\Cache\Adapter\ArrayAdapter;
+
+    $cache = new ArrayAdapter(
+        // in seconds; applied to cache items that don't define their own lifetime
+        // 0 means to store the cache items indefinitely (i.e. until the current PHP process finishes)
+        $defaultLifetime = 0,
+        // if ``true``, the values saved in the cache are serialized before storing them
+        $storeSerialized = true
+    );
+
+Filesystem Cache Adapter
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This adapter is useful when you want to improve the application performance but
+can't install tools like APC or Redis in the server. This adapter stores the
+contents as regular files in a set of directories on the local file system::
+
+    use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+
+    $cache = new FilesystemAdapter(
+        // the subdirectory of the main cache directory where cache items are stored
+        $namespace = '',
+        // in seconds; applied to cache items that don't define their own lifetime
+        // 0 means to store the cache items indefinitely (i.e. until the files are deleted)
+        $defaultLifetime = 0,
+        // the main cache directory (the application needs read-write permissions on it)
+        // if none is specified, a directory is created inside the system temporary directory
+        $directory = null
+    );
+
+APCu Cache Adapter
+~~~~~~~~~~~~~~~~~~
+
+This adapter can increase the application performance very significantly, because
+contents are cached in the memory of your server, which is much faster than the
+file system. It requires to have installed and enabled the PHP APC extension.
+It's not recommended to use it when performing lots of write and delete
+operations because it produces fragmentation in the APCu memory and that
+degrades performance significantly::
+
+    use Symfony\Component\Cache\Adapter\ApcuAdapter;
+
+    $cache = new ApcuAdapter(
+        // the string prefixed to the keys of the items stored in this cache
+        $namespace = '',
+        // in seconds; applied to cache items that don't define their own lifetime
+        // 0 means to store the cache items indefinitely (i.e. until the APC memory is deleted)
+        $defaultLifetime = 0
```

Comment:
If we have a namespace, why do we need this "nonce" option? Looks like a "subnamespace". Another question, if it's the "version", why is called "nonce" instead of "version"? (By the way, "nonce" in English mean "now, today, hear and now, etc.)